Sources/NIO/BaseSocketChannel.swift

```diff
@@ -755,9 +751,36 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
         }
     }
 
+    final func readEOF() {
+        if self.lifecycleManager.isRegistered {
+            self.safeReregister(interested: self.interestedEvent.subtracting(.readEOF))
+
+            while self.lifecycleManager.isActive {
```

Comment:
Can we ever drop out of the loop due to a change in `isActive` without hitting `didReceiveEOF`?

Reply:
@Lukasa yes, if the users closes in `channelRead`. Then we currently just discard the remaining ones. Adding a test.

Reply:
@normanmaurer / @Lukasa is that what we want?